src/palm-react/hooks/app/useAlphaTest.ts

```diff
@@ -0,0 +1,98 @@
+import { UTIL } from 'palm-core/libs'
+import { useEffect, useState } from 'react'
+
+type AlphaConfig = {
+  waitlist: boolean
+}
+
+/* Hook used only in alpha test
+ * 1. fetch config.json from github assets
+ * 2. fetch waitlist.json from github assets
+ * 3. return config and waitlist
+ */
+const useAlphaTest = (): {
+  config: AlphaConfig | undefined
+  waitList: string[] | undefined
+} => {
+  const [config, setConfig] = useState<AlphaConfig | undefined>(undefined)
+  const [waitList, setWaitList] = useState<string[] | undefined>(undefined)
+
+  const fetchWithTimeout = async (
+    url: string,
+    timeout: number
+  ): Promise<Response> => {
+    const controller = new AbortController()
+    const timeoutId = setTimeout(() => controller.abort(), timeout)
+
+    try {
+      const response = await fetch(url, { signal: controller.signal })
+      clearTimeout(timeoutId)
+      return response
+    } catch (error) {
+      clearTimeout(timeoutId)
+      throw error
+    }
+  }
+
+  const fetchConfig = async (): Promise<void> => {
+    try {
+      const ret = await fetchWithTimeout(
+        'https://raw.githubusercontent.com/palmapp-xyz/assets/main/mobile/alpha/config.json',
```

Comment:
what is config vs waitlist?

Reply:
config.json -> whether to use waitlist
waitlist.json -> waitlist address